notifications/notification.py

```diff
@@ -33,212 +37,253 @@
   Der Hinweistext ist frei wählbar.
 """
 
-with open('mail_pw', 'r') as pw:
-    mail_pw = pw.read().replace('\n', '')
 
-def send_notification(to_address, subject, message):
-    msg = MIMEText(message,_charset='utf-8')
+def send_notification(to_address, subject, content_text, content_html):
+    msg = MIMEMultipart('alternative')
+
+    plain = MIMEText(content_text, 'plain', _charset='utf-8')
+    html = MIMEText(content_html, 'html', _charset='utf-8')
 
-    msg['Subject'] = subject
-    msg['From'] = MAIL_FROM
+    msg.attach(plain)
+    msg.attach(html)
+
+    msg['Subject'] = SUBJECT_PREFIX + " " + subject
+    msg['From'] = config.MAIL_FROM
     msg['To'] = to_address
     msg['Date'] = formatdate(time.time(), localtime=True)
     msg['Message-id'] = make_msgid()
 
     logging.info("Mailing %s: '%s'", to_address, subject)
-    logging.debug("Content: ---\n%s\n---", message)
+    logging.debug("Content: ---\n%s\n---", content_text)
     s = smtplib.SMTP()
-    s.connect(host='vega.uberspace.de', port=587)
+    s.connect(host=config.MAIL_HOST, port=config.MAIL_PORT)
     s.ehlo()
     s.starttls()
-    s.login(user=MAIL_FROM, password=mail_pw)
-    s.sendmail(MAIL_FROM, [to_address], msg.as_string())
+    s.login(user=config.MAIL_FROM, password=config.MAIL_PW)
+    s.sendmail(config.MAIL_FROM, [to_address], msg.as_string())
     s.quit()
 
-def send_drink(user, drink, balance):
+
+def send_drink(user, drink, with_summary=False):
     try:
-        user_email = user['email']
-
-        if user_email and user['meta']['drink_notification'] == 'instant':
-            mail_msg = ("Du hast das folgende Getränk getrunken {drink_name}" \
-                       "\n\nVerbleibendes Guthaben: EUR {balance}"+ \
-                       FOOTER).format(
-                drink_name=drink['name'], balance=balance, uid=user['id'])
-            send_thread = threading.Thread(
-                target=send_drink_with_summary,
-                args=(user, "Getränk getrunken", mail_msg)
-            )
-            send_thread.daemon = True
-            send_thread.start()
+        if user['meta']['drink_notification'] == 'instant':
```

Comment:
check for `if user["email"]` whether the user has a mail address

Reply:
also... because of the new "instant and daily", the check should be
`if 'user['email'] and 'instant' in user['meta']['drink_notification']:`